impress-org/givewp-next-gen
# Feature: Add Donation Form Modeling

## Description



This PR adds a `DonationForm` model with a corresponding repository and factory.

The builder's resource controller is now using this model to show and update the form.

One nice feature of the form model is its representation of blocks as a type of `BlockCollection`.  This allows us to interact with the form's blocks as a specific type, rather than an ambiguous string.

## Testing Instructions



Pull branch and build assets.

Click the "Visual Builder (Alpha)" button in the admin to make sure it still generates a form correctly.
Interact with the form builder, changing settings & fields and make sure everything saves properly.

You can also use our new factory to generate a form that will work in the admin!

```
$form = DonationForm::factory()->create();
$form->blocks; // BlockCollection
$form->getSchema() // fields api from block collection
```

## Pre-review Checklist
